resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -200,6 +207,32 @@ public DnsNameResolver(
         this.searchDomains = checkNotNull(searchDomains, "searchDomains").clone();
         this.ndots = checkPositiveOrZero(ndots, "ndots");
 
+        boolean cnameFollowARecords = false;
+        boolean cnameFollowAAAARecords = false;
+        // Use LinkedHashSet to maintain correct ordering.
+        Set<DnsRecordType> recordTypes = new LinkedHashSet<DnsRecordType>(resolvedAddressTypes.length);
```

Comment:
Let me leave it as it is for now... If you really feel strong we can do a follow up.